XamarinCommunityToolkit/Effects/EntryFormattedTextEffect.android.cs

```diff
@@ -0,0 +1,258 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Linq;
+using Android.Text;
+using Android.Text.Style;
+using Android.Widget;
+using Xamarin.CommunityToolkit.Android.Effects;
+using Xamarin.CommunityToolkit.Effects;
+using Xamarin.Forms;
+using Xamarin.Forms.Platform.Android;
+
+[assembly: ResolutionGroupName(nameof(Xamarin.CommunityToolkit))]
+[assembly: ExportEffect(typeof(EntryFormattedTextEffect), nameof(EntryFormattedTextEffect))]
+
+namespace Xamarin.CommunityToolkit.Android.Effects
+{
+	/// <summary>
+	/// The FormattedText <see cref="PlatformEffect"/> specific
+	/// to <see cref="Entry"/> controls.
+	/// </summary>
+	public class EntryFormattedTextEffect : PlatformEffect, IDisposable
+	{
+		bool isDisposed;
+
+		protected IList<FormattedTextSegment<Java.Lang.Object>> Segments { get; set; }
+
+		/// <summary>
+		/// Gets the native <see cref="FormsEditText"/> control.
+		/// </summary>
+		public new FormsEditText Control => (FormsEditText)base.Control;
+
+		/// <summary>
+		/// Gets the Xamarin.Forms <see cref="Entry"/> element.
+		/// </summary>
+		public new Entry Element => (Entry)base.Element;
+
+		/// <inheritdoc />
+		protected override void OnAttached()
+		{
+			UpdateFormattedText();
+
+			Control.TextChanged += OnTextChanged;
+		}
+
+		/// <inheritdoc />
+		protected override void OnDetached()
+		{
+		}
+
+		/// <inheritdoc />
+		protected override void OnElementPropertyChanged(PropertyChangedEventArgs args)
+		{
+			base.OnElementPropertyChanged(args);
+
+			if (args.PropertyName == FormattedTextEffect.FormattedTextProperty.PropertyName)
+				UpdateFormattedText();
+		}
+
+		void OnTextChanged(object sender, global::Android.Text.TextChangedEventArgs args)
+		{
```

Comment:
The code looks very code to me! I found just one place that we can improve perf a little bit. Accessing prop values directly is bad, we can use a local variable instead. So in this case `args.Start` is used more than once, so would be good to do something like `var argStart = args.Start` and use this `argStart` in the code. Since this method will be called a lot, this can have a good impact.
Yesterday night I tried to set up a benchmark for this feature but without success. I'll try again in the next days and share it with you.
And awesome work!

Reply:
Thank you 🙏 for taking a look at this. Anything we can do to improve the overall performance of this is important to get right in this algorithm.